spring-integration-core/src/main/java/org/springframework/integration/context/IntegrationObjectSupport.java

```diff
@@ -143,9 +142,15 @@ public void setChannelResolver(DestinationResolver<MessageChannel> channelResolv
 
 	@Override
 	public final void afterPropertiesSet() {
+		this.integrationProperties = IntegrationContextUtils.getIntegrationProperties(this.beanFactory);
 		try {
 			if (this.messageBuilderFactory == null) {
-				this.messageBuilderFactory = IntegrationUtils.getMessageBuilderFactory(this.beanFactory);
+				if (this.beanFactory != null) {
```

Comment:
Shouldn't we just move this to `getMessageBuilderFactory()` to make it really lazy.